File: 백준/Gold/1863. 스카이라인 쉬운거/스카이라인 쉬운거.java
import java.io.IOException;
import java.util.Stack;
import java.util.StringTokenizer;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int total = 0;
        Stack<Integer> stack = new Stack<>();
        stack.push(0);
        
        int n = Integer.parseInt(st.nextToken());
        
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            
            while (stack.size() > 0 && stack.lastElement() > y) {
                total += 1;
                stack.pop();
            }
            
            if (stack.lastElement() < y) {
                stack.push(y);
            }
        }

        System.out.println(total + stack.size() - 1);
    }
}

File: 백준/Gold/1863. 스카이라인 쉬운거/스카이라인 쉬운거.py
# 스카이라인 쉬운거
import sys
input = sys.stdin.readline

n = int(input())

stack = [0]
total = 0

for i in range(n):
    x, y = map(int,input().split())

    while stack and stack[-1] > y:
        total += 1
        stack.pop()

    if stack[-1] < y:
        stack.append(y)

print(total+len(stack)-1)
